enterprise/plugins/com.liferay.ide.server.websphere.core/src/com/liferay/ide/server/websphere/util/WebsphereUtil.java

```diff
@@ -0,0 +1,1106 @@
+    {
+        if( ( filePath == null ) || ( filePath.length() == 0 ) )
+        {
+            return false;
+        }
+        String testPath = System.getenv( "LOCALAPPDATA" );
+
+        if( ( testPath == null ) || ( testPath.length() == 0 ) )
+        {
+            return false;
+        }
+        testPath = testPath + "\\VirtualStore";
+        int i = filePath.indexOf( ":" );
+
+        if( i != -1 )
+        {
+            if( i + 1 != filePath.length() )
+            {
+                testPath = testPath + filePath.substring( i + 1 );
+            }
+        }
+        else
+        {
+            testPath = testPath + filePath;
+        }
+
+        File tempFile = new File( testPath );
+        return tempFile.exists();
+    }
+
+    public static boolean isPortAvailable( int port )
+    {
+        try
+        {
+            bindPort( "0.0.0.0", port );
+            bindPort( InetAddress.getLocalHost().getHostAddress(), port );
+            return true;
+        }
+        catch( Exception e )
+        {
+            return false;
+        }
+    }
+
+    public static boolean isWebsphereRunning( int pid )
+    {
+        if( pid <= 0 )
+        {
+            return false;
+        }
+        String[] cmd = null;
+
+        boolean isRunning = false;
+
+        if( CoreUtil.isWindows() )
+        {
+            cmd = new String[] { "tasklist" };
+        }
+        else if( CoreUtil.isLinux() )
+        {
+            cmd = new String[] { "ps", "-ef" };
+        }
+        else
+        {
+            return false;
+        }
+
+        String pidVal = String.valueOf( pid );
+        int len = pidVal.length();
+        if( cmd != null )
+        {
+
+        }
+        try(BufferedReader pidBuffer =
+            new BufferedReader( new InputStreamReader( Runtime.getRuntime().exec( cmd ).getInputStream() ) ))
+        {
+            String line = null;
+            while( ( line = pidBuffer.readLine() ) != null )
+            {
+                int pos = line.indexOf( pidVal );
+
+                if( pos == -1 )
+                {
+                    continue;
+                }
+
+                String charBefore = line.substring( pos - 1, pos );
+                String charAfter = line.substring( pos + len, pos + len + 1 );
+
+                if( ( charBefore.equals( " " ) ) && ( charAfter.equals( " " ) ) )
+                {
+                    isRunning = true;
+                }
+            }
+        }
+        catch( IOException e )
+        {
+            WebsphereCore.logError( e );
+        }
+
+        return isRunning;
+    }
+
+    public static boolean isWebsphereRuntime( IRuntime runtime )
+    {
+        return runtime != null &&
+            runtime.getRuntimeType().getId().contains( "com.liferay.ide.eclipse.server.websphere.runtime" ) &&
+            getWebsphereRuntime( runtime ) != null;
+    }
+
+    public static void killWebsphereProcess( int pid )
```

Comment:
The stopServer.sh does I have followed it, but the server sometimes didn't release jmx port even if we invoke stop. This will make sure all the ports used by websphere will be released when the server stopped. I referenced this IBM Eclipse webspehre tools.